x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * A connector document consists of:
+ * <ul>
+ *     <li>...</li>
+ * </ul>
+ */
+public class Connector implements Writeable, ToXContentObject {
+
+    private final String connectorId;
+
+    @Nullable
+    private final String apiKeyId;
+    @Nullable
+    private final Map<String, Object> configuration;
```

Comment:
What is configuration composed of? Is `Map<String,Object>` necessary here or should this be more structured?

Reply:
++ for adding explicit types!

Reply:
Is it acceptable to defer defining explicit types until the next PR? Regarding the 'update configuration' endpoint, there is a proposal to separate the configuration schema from its values. I believe it would simplify the review process if I tackled this in a dedicated PR alongside the update configuration endpoint changes.

Reply:
I think that's fair @jedrazb . Just make a note / TODO so we understand there's follow up work on this 💯